unity-client/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/Gif/GifProcessor/GifProcessor.cs

```diff
@@ -48,9 +49,9 @@ public void DisposeGif()
         {
             DCL.GIFProcessingBridge.i.DeleteGIF(url);
         }
-        else if (!(webRequest is null))
+        else if (webRequestOp != null)
```

Comment:
The previous logic nulls the webRequest to make sure it isn't running in this statement. This is an ongoing pattern in all the project, so please be extra careful here. We should keep the logic and null the webRequestOp below or check for !IsDisposed().

Reply:
Notice that the `Dispose()` we are using here belongs to our custom `WebRequestAsyncOperation` and it is already nulling the request:

![image](https://user-images.githubusercontent.com/64659061/114906568-74274100-9e1a-11eb-8a08-51b60b0e684b.png)

Reply:
The `if` checks against the `webRequestOp` to check if the request is in progress. If that's true then the if should be `if (webRequestOp?.webRequest != null)` or something like that.

Reply:
If we thing about this specific case, what it should happen if we reach this `if` is that, whether the request is in progress or not, we should dispose it (and abort it before if it is in progress in that moment), and our new `Dispose()` is already doing that:

![image](https://user-images.githubusercontent.com/64659061/114997251-ee4dd900-9e9f-11eb-8e88-42a080a9b804.png)
![image](https://user-images.githubusercontent.com/64659061/114998886-a7f97980-9ea1-11eb-89c4-1cbcf642c330.png)

Maybe in the old code, using directly the `UnityWebRequest`, we always had to check this kind of things before an abort or a dispose, but now we have a `WebRequestAsyncOperation` that acts as a "container" of the request and we delegate all these kind of checking responsabilities to its internal logic (in functions like `Abort()`, `Dispose()`, etc.). That is why we only make the null-check of the "container" now, the responsability of the rest of checkings has been moved to the own request operation. This allows us to avoid always having to think about what checks to make before execute an abort or a dispose, we simply order the specific operation to `WebRequestAsyncOperation` and he will do the rest.

Don't worry, this disposing case is under control.